ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py

```diff
@@ -1015,3 +1015,51 @@ def test_expand(
         x=x[0],
         shape=shape,
     )
+
+
+@st.composite
+def _concat_from_sequence_helper(draw):
+    shape = draw(st.shared(helpers.get_shape(min_num_dims=1), key="values_shape"))
+    num_arrays = draw(
+        st.shared(helpers.ints(min_value=1, max_value=3), key="num_arrays")
+    )
+    dtype = draw(st.sampled_from(draw(helpers.get_dtypes("valid"))))
+    arrays = []
+    dtypes = [dtype for _ in range(num_arrays)]
+
+    for _ in range(num_arrays):
+        array = draw(helpers.array_values(dtype=dtype, shape=shape))
+        arrays.append(np.asarray(array, dtype=dtype))
```

Comment:
You don't need to do this to create a sequence of arrays. A more efficient and simple way would be to use the `dtype_and_values` helper where you can set the `num_arrays` argument. You can also set `ret_shape` argument to get back the shape of the arrays.